miner/worker.go

```diff
@@ -919,50 +921,64 @@ func (w *worker) commitNewWork(interrupt *int32, noempty bool, timestamp int64)
 
 	w.updateSnapshot()
 
-	// Play our part in generating the random beacon.
-	if w.isRunning() && random.IsRunning() {
-		istanbul, ok := w.engine.(consensus.Istanbul)
-		if !ok {
-			log.Crit("Istanbul consensus engine must be in use for the randomness beacon")
+	if w.isRunning() {
+		// TODO use caller instead of parentCaller (HF required)
+		randomIsActive := false
+		header := w.chain.CurrentHeader()
+		state, err := w.chain.StateAt(header.Root)
+		// TODO(HF) make this check use w.current.header instead of blockchain's current
```

Comment:
Same comment as elsewhere as to whether this is really a HF and whether there is a reason to keep it this way.

Reply:
Changed to use vmRunner from current header & state